network_analysis/nbconverted/2A_community_detection_planted_partition.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Run community detection experiments for planted partition model
+# 
+# This script runs community detection on the eADAGE correlation network using the [planted partition model](https://en.wikipedia.org/wiki/Stochastic_block_model#Special_cases). Here, we use the implementation in the [graph-tool package](https://graph-tool.skewed.de/static/doc/demos/inference/inference.html#assortative-community-structure).
+# 
+# Using the planted partition model rather than the full stochastic block model ensures that communities are assortative (i.e. more connections within than across communities), which is what we're looking for in this analysis.
+
+# In[1]:
+
+
+import os
+
+import numpy as np
+import pandas as pd
+import graph_tool.all as gt
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches 
+import seaborn as sns
+
+from sample_nodes import (
+    sample_degree_matched,
+    sort_by_degree,
+)
+
+gt.seed_rng(1)
+np.random.seed(1)
+
+
+# In[2]:
+
+
+# relevant file paths
+data_dir = './data'
+processed_graph = os.path.join(data_dir, 'eadage_generic_graph_unsigned.gt')
+
+# script parameters
+NUM_NODE_SAMPLES = 1000 # number of degree-matched node samples for permutation test
+NUM_BINS = 100 # number of bins to divide nodes into, for sampling
+
+
+# In[3]:
+
+
+G = gt.load_graph(processed_graph)
+# make sure vertex/edge properties exist
+print(G)
+print(list(G.vp.keys()))
+print(list(G.ep.keys()))
+
+
+# In[4]:
+
+
+# community detection using planted partition model
+state = gt.PPBlockState(G)
```

Comment:
Does state include a membership assignment mapping?

Reply:
It does, once the model is fit (which is what the line below this does). See [here](https://graph-tool.skewed.de/static/doc/inference.html#graph_tool.inference.blockmodel.BlockState) for more details (the "block state" is the membership assignments).